### ISSUE TITLE
handle unresponsive server

### ISSUE THREAD

**Issue by obayemi** [CleverCloud/lapin#333]
When connecting to un unresponsive server (e.g. docker proxy that didn't yet kill the connection on server restart) lapin goes on an infinite loop 

Code example:
```
use lapin::{Connection, ConnectionProperties};
use tokio_amqp::*;

const AMQP_URL : &str = "amqp://foo:bar@localhost:4242//";

#[tokio::main]
async fn main() {
    env_logger::init();
    println!("Hello, world!");
    log::debug!("bootstrap amqp");
    let conn = Connection::connect(AMQP_URL, ConnectionProperties::default().with_tokio()).await.unwrap();
    log::debug!("established amqp connection");
    let channel = conn.create_channel().await.unwrap();
    log::debug!("created amqp channel");
}
```
then open a connection with `nc -l -p 4242`, if you drop the connection with ctrl-d before sending the first Byte, lapin goes on an infinite loop of 
```
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
[2021-04-26T14:15:26Z ERROR lapin::io_loop] Socket was readable but we read 0, marking as wouldblock
```

**Comment by obayemi** [comment]
I'm not sure that I really understand fully how the lib works, but in
https://github.com/CleverCloud/lapin/blob/7035e7a107198bc0c063b94b10969cdde42daf44/src/io_loop.rs#L342-L348
if the io loop receives 0 bytes it doesn't seem to register it, without making a difference between reading 0 bytes and not having data to read (where it would make sense to mark as wouldblock)

**Comment by dbcfd** [comment]
@obayemi Can you try #335 and see if it fixes it for you?

**Comment by Keruspe** [comment]
Fixed in 1.7.1
